src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/SqlConnectionX.cs

```diff
@@ -313,6 +333,35 @@ protected override DbBatch CreateDbBatch()
         /// <inheritdoc/>
         protected override DbCommand CreateDbCommand()
             => throw new NotImplementedException();
+
+        #region internal helpers
+
+        internal void OnInfoMessage(SqlInfoMessageEventArgs imevent, out bool notified)
+        {
+            // TODO review event source traces later
+            // SqlClientEventSource.Log.TryTraceEvent("SqlConnection.OnInfoMessage | API | Info | Object Id {0}, Message '{1}'", ObjectID, imevent.Message);
+            SqlInfoMessageEventHandler? handler = InfoMessage;
+            if (null != handler)
+            {
+                notified = true;
+                try
+                {
+                    handler(this, imevent);
+                }
+                catch (Exception e)
```

Comment:
Are exception filters allowed? They might make it more readable.

Reply:
The SqlInfoMessageEventHandler is implemented by user, so we capture generic exception here.